web-wallet/src/lib/containers/MigrateContract/MigrateContract.svelte

```diff
@@ -0,0 +1,468 @@
+<script>
+  import { AppAnchor, AppImage } from "$lib/components";
+  import {
+    Button,
+    ExclusiveChoice,
+    Icon,
+    Textbox,
+    Wizard,
+    WizardStep,
+  } from "$lib/dusk/components";
+  import {
+    mdiArrowRight,
+    mdiCheckDecagramOutline,
+    mdiInformationOutline,
+    mdiTimerSand,
+    mdiWalletOutline,
+  } from "@mdi/js";
+  import { createCurrencyFormatter } from "$lib/dusk/currency";
+  import { logo } from "$lib/dusk/icons";
+  import { calculateAdaptiveCharCount, middleEllipsis } from "$lib/dusk/string";
+  import { onMount } from "svelte";
+  import { settingsStore } from "$lib/stores";
+
+  const ERC_TOKEN = "ERC-20";
+  const BEP_TOKEN = "BEP-20";
+
+  const options = [
+    { disabled: false, label: ERC_TOKEN, value: ERC_TOKEN },
+    { disabled: false, label: BEP_TOKEN, value: BEP_TOKEN },
+  ];
+
+  /** @type {String} */
+  let selected = ERC_TOKEN;
+
+  /** @type {Boolean} */
+  const migrationInProgress = false;
+
+  /** @type {Boolean} */
+  let isWalletConnected = false;
+
+  /** @type {String} */
+  let connectedWalletAddress =
+    "6rmam3FisEHih84oJMEsmi2GGVetDPmcjw7H4heeVbm79DwZUPvPJTBeFriLx85Cy2Q2cGisz32BeSYX99kNXjD";
+
+  /** @type {Number} */
+  let connectedWalletBalance = 10;
+
+  /** @type {Number|String} */
+  let amount = "";
```

Comment:
Why this can be both number and string?

It's quite confusing (as are confusing the checks you make afterwards to see if it's a number).

Reply:
Changed the type from `string` to `null`, needed something else than a number for the `amount` as the field must show the placeholder first with no value.